django_lti_tool_provider/views.py

```diff
@@ -0,0 +1,148 @@
+from django.views.generic import View
+import oauth2
+import logging
+
+from django.conf import settings
+from django.core.urlresolvers import reverse, NoReverseMatch
+from django.http import HttpResponseRedirect, HttpResponseBadRequest
+from django.views.decorators.csrf import csrf_exempt
+
+from ims_lti_py.tool_provider import DjangoToolProvider
+
+from django_lti_tool_provider.models import LtiUserData
+from django_lti_tool_provider.signals import Signals
+
+
+_logger = logging.getLogger(__name__)
+
+
+class LTIView(View):
+    authentication_hooks = []
+
+    PASS_TO_AUTHENTICATION_HOOK = {
+        'lis_person_sourcedid': 'username',
+        'lis_person_contact_email_primary': 'email',
+        'user_id': 'user_id'
+    }
+
+    @csrf_exempt
+    def dispatch(self, *args, **kwargs):
+        return super(LTIView, self).dispatch(*args, **kwargs)
+
+    def get(self, request, *args, **kwargs):
+        return self.process_request(request)
+
+    def post(self, request, *args, **kwargs):
+        return self.process_request(request)
+
+    def process_request(self, request):
+        if not request.user.is_authenticated():
+            try:
+                lti_parameters = self._get_lti_parameters_from_request(request)
+                lti_data = {
+                    hook_name: lti_parameters.get(lti_name, None)
```

Comment:
@smarnach to isolate client of the fact that it is an LTI request, so that `lis_person_contact_email_primary` are not leaked into "client" code. On the second thought, I think it would make sense to pass it as keywords arguments, not dictionary, so that client hook signature would look like

    def auth_hook(user_id, user_name, email):